src/Microsoft.OData.Core/Uri/ODataUriConversionUtils.cs

```diff
@@ -536,14 +550,23 @@ private static void WriteStartResource(ODataWriter writer, ODataResourceBase res
         /// <param name="writeValue">Delegate to use to actually write the value.</param>
         private static void WriteJsonLightLiteral(IEdmModel model, ODataMessageWriterSettings messageWriterSettings, TextWriter textWriter, Action<ODataJsonLightValueSerializer> writeValue)
         {
+
+            IEnumerable<KeyValuePair<string,string>> parameters = new Dictionary<string, string>
+            {
+                {MimeConstants.MimeIeee754CompatibleParameterName, messageWriterSettings.IsIeee754Compatible.ToString() }
+            };
+            ODataMediaType mediaType = new ODataMediaType(MimeConstants.MimeApplicationType, MimeConstants.MimeJsonSubType, parameters);
+
             // Calling dispose since it's the right thing to do, but when created from a custom-built TextWriter
             // the output context Dispose will not actually dispose anything, it will just cleanup itself.
             // TODO: URI parser will also support DI container in the future but set the container to null at this moment.
             ODataMessageInfo messageInfo = new ODataMessageInfo
             {
                 Model = model,
                 IsAsync = false,
-                IsResponse = false
+                IsResponse = false,
+                MediaType = mediaType

```

Comment:
I'd go with only adding the header + MediaType only when it's necessary - in this case only when `IsIeee754Compatible == true` instead of always having it there. No need adding response headers when not required e.g. when `IsIeee754Compatible  == false`

Reply:
I think it is good to be explicit especially when the header has been provided.